tests/int_key.rs

```diff
@@ -0,0 +1,96 @@
+#![allow(dead_code, unused)]
+use std::collections::HashMap;
+
+use serde::Deserialize;
+use yaml_rust::YamlLoader;
+use yaml_rust::Yaml;
+
+use config::Config;
+
+#[derive(Debug, Deserialize)]
+struct Outer {
+    inner_string: HashMap<String, Inner>,
+    inner_int: HashMap<u32, Inner>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Inner {
+    member: String,
+}
+
+const CONFIG: &str = r#"
+inner_int:
+  "1":
+    member: "Test Int 1"
+  2:
+    member: "Test Int 2"
+inner_string:
+  stuff:
+    member: "Test String"
+"#;
+
+#[test]
+fn test_int_key() {
+    let config = Config::builder()
+        .add_source(config::File::from_str(CONFIG, config::FileFormat::Yaml))
+        .build()
+        .unwrap();
+
+    let outer: Outer = config.try_deserialize().unwrap();
+    assert_eq!(outer.inner_string.get("1").unwrap().member, "Test Int 1");
```

Comment:
I don't know whether the key `"1"` being parsed into an int is actually correct behaviour, I'm not very familiar with yaml. However, it seems this is the case currently.

Anyway, assuming that behaviour is wanted - this test fails because it gets from the wrong map, and when getting from the correct map it expects a `&u32` instead of a `&str`.

```suggestion
    assert_eq!(outer.inner_int.get(&1).unwrap().member, "Test Int 1");
```